pkg/tsdb/legacydata/service/service.go

```diff
@@ -93,37 +146,27 @@ func (h *Service) HandleRequest(ctx context.Context, ds *models.DataSource, quer
 			JSON:      modelJSON,
 		})
 	}
+	return req, nil
+}
 
-	resp, err := h.pluginsClient.QueryData(ctx, req)
-	if err != nil {
-		return legacydata.DataResponse{}, err
-	}
-
-	tR := legacydata.DataResponse{
-		Results: make(map[string]legacydata.DataQueryResult, len(resp.Responses)),
+func customHeaders(jsonData *simplejson.Json, decryptedJsonData map[string]string) map[string]string {
+	if jsonData == nil {
+		return nil
 	}
 
-	for refID, r := range resp.Responses {
-		qr := legacydata.DataQueryResult{
-			RefID: refID,
-		}
+	data := jsonData.MustMap()
 
-		for _, f := range r.Frames {
-			if f.RefID == "" {
-				f.RefID = refID
+	headers := map[string]string{}
+	for k := range data {
+		if strings.HasPrefix(k, headerName) {
+			if header, ok := data[k].(string); ok {
+				valueKey := strings.ReplaceAll(k, headerName, headerValue)
+				headers[header] = decryptedJsonData[valueKey]
```

Comment:
Isn't `valueKey` just `fmt.Sprintf("httpHeaderValue%d", i)` where i is the index of the loop, like in ngalert?

Reply:
I can't actually find the code that writes down "httpHeaderValue" to the store anywhere in the go code. Looks like that's handled by the frontend code. I'm rather happy to use the map keys and extract whatever suffix string happens to exist, especially since we don't have any guarantees that the frontend won't change this.